File: app/src/main/java/com/platdmit/simplecloudmanager/app/helpers/charts/serverValueFormatter.java
package com.platdmit.simplecloudmanager.app.helpers.charts;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.data.BubbleEntry;
import com.github.mikephil.charting.data.CandleEntry;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.PieEntry;
import com.github.mikephil.charting.data.RadarEntry;
import com.github.mikephil.charting.formatter.ValueFormatter;

import java.util.Map;

public class serverValueFormatter extends ValueFormatter {
    private final Map<Float, String> mTitles;
    public serverValueFormatter(Map<Float, String> titles) {
        mTitles = titles;
    }

    @Override
    public String getAxisLabel(float value, AxisBase axis) {
        return mTitles.get(value);
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/utilities/charts/ServerValueFormatter.kt
package com.platdmit.simplecloudmanager.utilities.charts

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.ValueFormatter

class ServerValueFormatter(private val titles: Map<Float, String>) : ValueFormatter() {
    override fun getAxisLabel(value: Float, axis: AxisBase): String {
        return titles[value] ?: ""
    }
}